test/e2e/tests/test_tenant_model_inference.py

```
@@ -274,3 +279,149 @@ def test_tenant_isolation_cross_gateway_blocked(self, tenant_inference_cases):
                 f"Unexpected API key creation failure: {api_key_response.status_code} "
                 f"{redact_sensitive(api_key_response.text)}"
             )
+
+
+# ─── Body-based routing ────────────────────────────────────────────────────
+
+
+def _check_payload_processing_pods():
+    """Check if payload-processing (IPP) pods are deployed."""
+    for name in ("payload-pre-processing", "payload-processing"):
+        result = subprocess.run(
```

Ruff (0.15.20)

[error] `subprocess` call: check for execution of untrusted input

(S603)